staging/src/k8s.io/client-go/transport/token_source_test.go

```diff
@@ -154,3 +155,67 @@ func TestCachingTokenSourceRace(t *testing.T) {
 		}
 	}
 }
+
+func TestCancellation(t *testing.T) {
+	tests := []struct {
+		name          string
+		header        http.Header
+		wrapTransport func(http.RoundTripper) http.RoundTripper
+	}{
+		{
+			name: "cancel req with bearer token skips oauth rt",
+
+			header: map[string][]string{"Authorization": {"Bearer TOKEN"}},
+		},
+		{
+			name: "cancel req without bearer token hits both rts",
+		},
+		{
+			name: "cancel req without bearer token hits both wrapped rts",
+			wrapTransport: func(rt http.RoundTripper) http.RoundTripper {
+				return NewUserAgentRoundTripper("testing testing", rt)
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			baseRecorder := &recordCancelRoundTripper{}
+
+			var base http.RoundTripper = baseRecorder
+			if test.wrapTransport != nil {
+				base = test.wrapTransport(base)
+			}
+
+			rt := &tokenSourceTransport{
+				base: base,
+				ort: &oauth2.Transport{
+					Base: base,
+				},
+			}
+
+			rt.CancelRequest(&http.Request{
+				Header: test.header,
+			})
+
+			if baseRecorder.canceled == false {
```

Comment:
could you add a test case 'unable to cancel request'?

Reply:
Done.